File: introducao-a-computacao/2022/trabalho-pratico/csharp/domain/ContadorParesImpares.cs
using System.Collections.Generic;

namespace IntroComp.Domain
{
    public static class ContadorParesImpares
    {
        public static ResultadoContagem Contar(IReadOnlyList<int> numeros)
        {
            if (numeros == null)
                throw new System.ArgumentNullException(nameof(numeros));

            int pares = 0, impares = 0;
            for (int i = 0; i < numeros.Count; i++)
            {
                if (numeros[i] % 2 == 0)
                    pares++;
                else
                    impares++;
            }
            return new ResultadoContagem(pares, impares);
        }

        public static string FormatarResultado(ResultadoContagem r)
        {
            return "\n===== Resultado =====\n" +
                   $"Quantidade de números pares: {r.Pares}\n" +
                   $"Quantidade de números ímpares: {r.Impares}";
        }
    }
}

File: introducao-a-computacao/2022/trabalho-pratico/java/src/introcomp/domain/ContadorParesImpares.java
package introcomp.domain;

import java.util.Objects;


public final class ContadorParesImpares {

    private ContadorParesImpares() {
    }

    public static ResultadoContagem contar(int[] numeros) {
        Objects.requireNonNull(numeros, "numeros");
        int pares = 0, impares = 0;
        for (int n : numeros) {
            if (n % 2 == 0) {
                pares++;
            } else {
                impares++;
            }
        }
        return new ResultadoContagem(pares, impares);
    }

    public static String formatarResultado(ResultadoContagem r) {
        Objects.requireNonNull(r, "resultado");
        return """
                ===== Resultado =====
                Quantidade de números pares: %d
                Quantidade de números ímpares: %d""".formatted(r.pares(), r.impares());
    }
}
